language/move-prover/src/prelude.bpl

```diff
@@ -132,7 +132,7 @@ function {:inline} $IsValidU128Vector(vec: Value): bool {
 }
 
 function {:inline} $IsValidNum(v: Value): bool {
-  is#Integer(v) && i#Integer(v) >= 0
+  is#Integer(v)
```

Comment:
Don't we want num to be non-negative?

Reply:
When I debugged your example, I verified that indeed sum_of_coins becomes negative. Only because the current use of `where` in spec variables actually doesn't guarantee the constraint is maintained, we did not see an assertion failure. Since we don't know a way yet to prevent this value becoming negative, I figured is it better we assume it is ok to do so, until we have a better idea.

Methodologically this means that one _always_ should use `num` and not `u64` or such for global spec vars for which the initial value is not known.